File: src/web/boss/src/api/hospital.ts
import axios from 'axios'
import { Pager, ValueObject } from '@/api/common'

// 业务医院 Hospital

/**
 * 业务医院交互模型
 */
export interface Hospital {
  id: string
  ownershipType: string
  hospitalType: string
  hospitalLevel: string
  status: string
  insuranceCode: string
  usccCode: string
  hospitalCode: string
  hospitalName: string
  provinceId: string
  province: string
  cityId: string
  city: string
  countyId: string
  county: string
  address: string
  postalCode: string
  longitude: number
  latitude: number
  mapUrl: string
  contactPhone: string
  contactEmail: string
  website: string
  summary: string
  description: string
  companionDiagnosisEnabled: boolean
  mealServiceEnabled: boolean
  testingDeliveryEnabled: boolean
}

/**
 * 业务医院视图模型
 */
export interface HospitalView {
  id: string
  ownershipType: ValueObject
  hospitalType: ValueObject
  hospitalLevel: ValueObject
  status: ValueObject
  insuranceCode: string
  usccCode: string
  hospitalCode: string
  hospitalName: string
  provinceId: string
  province: string
  cityId: string
  city: string
  countyId: string
  county: string
  address: string
  postalCode: string
  longitude: number
  latitude: number
  mapUrl: string
  contactPhone: string
  contactEmail: string
  website: string
  summary: string
  description: string
  companionDiagnosisEnabled: boolean
  mealServiceEnabled: boolean
  testingDeliveryEnabled: boolean
  createdAt: string
  updatedAt: string
  deletedAt: string
}

/**
 * 业务医院搜索模型
 */
export interface HospitalSearchModel {
  keywords: string
  hospitalCode: string
  deleted: boolean
}

/**
 * 获取业务医院
 * @param id
 */
export function getHospital(id: string) {
  return axios.get<HospitalView>(`/hospitals/${id}`)
}

/**
 * 搜索业务医院分页
 * @param data
 * @param page
 */
export function getHospitalPagination(data: HospitalSearchModel, page: Pager) {
  return axios.get<HospitalView>('/hospitals', { params: { ...data, ...page } })
}

/**
 * 添加业务医院
 * @param data
 */
export function createHospital(data: Hospital) {
  return axios.post('/hospitals', data)
}

/**
 * 修改业务医院信息
 * @param data
 */
export function updateHospital(data: Hospital) {
  return axios.put(`/hospitals/${data.id}`, data)
}

/**
 * 删除业务医院
 * @param id
 */
export function deleteHospital(id: string) {
  return axios.delete(`/hospitals/${id}`)
}

/**
 * 获取医院性质列表
 */
export function getHospitalLevelList() {
  return axios.get('/hospitals/levels')
}

/**
 * 获取医院所有制类型列表
 */
export function getHospitalOwnershipTypeList() {
  return axios.get('/hospitals/ownership-types')
}

/**
 * 获取医院类型列表
 */
export function getHospitalTypeList() {
  return axios.get('/hospitals/types')
}

File: src/server/core/src/main/java/com/zhenbanban/core/domain/internethospitalcontext/entity/Hospital.java
/*
 * Copyright (C) 2025 zhangxihai<mail@sniu.com>，All rights reserved.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 * WARNING: This code is licensed under the GPL. Any derivative work or
 * distribution of this code must also be licensed under the GPL. Failure
 * to comply with the terms of the GPL may result in legal action.
 */
package com.zhenbanban.core.domain.internethospitalcontext.entity;

import com.zhenbanban.core.domain.common.entity.AbsAggregate;
import com.zhenbanban.core.domain.internethospitalcontext.event.HospitalAddedEvent;
import com.zhenbanban.core.domain.internethospitalcontext.event.HospitalDestroyedEvent;
import com.zhenbanban.core.domain.internethospitalcontext.event.HospitalModifiedEvent;
import lombok.*;

import java.math.BigDecimal;

/**
 * 聚合根：业务医院表
 *
 * @author zhangxihai 2025/09/17
 */
@Data
@Builder(toBuilder = true)
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Hospital extends AbsAggregate {

    private Long id;

    private String ownershipType;

    private String hospitalType;

    private String hospitalLevel;

    private String status;

    private String insuranceCode;

    private String usccCode;

    private String hospitalCode;

    private String hospitalName;

    private String hospitalNamePinyin;

    private String hospitalNamePinyinAbbr;

    private Long provinceId;

    private String province;

    private Long cityId;

    private String city;

    private Long countyId;

    private String county;

    private String address;

    private String postalCode;

    private BigDecimal longitude;

    private BigDecimal latitude;

    private String mapUrl;

    private String contactPhone;

    private String contactEmail;

    private String website;

    private String summary;

    private String description;

    private Boolean companionDiagnosisEnabled;

    private Boolean mealServiceEnabled;

    private Boolean testingDeliveryEnabled;

    @Builder.Default
    private boolean deleted = false;

    /**
     * 添加
     */
    public void add() {
        this.setDeleted(false);
        this.setStatus("PENDING");
        this.setTestingDeliveryEnabled(false);

        HospitalAddedEvent event = HospitalAddedEvent.builder()
                .refId(this.getId())
                .hospitalId(this.getId())
                .hospitalName(this.getHospitalName())
                .build();

        this.addEvent(event);
    }

    /**
     * 修改
     */
    public void modify() {
        this.setDeleted(false);
        this.setStatus("PENDING");
        this.setTestingDeliveryEnabled(false);

        HospitalModifiedEvent event = HospitalModifiedEvent.builder()
                .refId(this.getId())
                .hospitalId(this.getId())
                .hospitalName(this.getHospitalName())
                .build();

        this.addEvent(event);
    }

    /**
     * 销毁
     */
    public void destroy() {
        this.setDeleted(true);

        HospitalDestroyedEvent event = HospitalDestroyedEvent.builder()
                .refId(this.getId())
                .hospitalId(this.getId())
                .hospitalName(this.getHospitalName())
                .build();

        this.addEvent(event);
    }

}
